src/api/music/albums.rs

```diff
@@ -154,12 +154,14 @@ async fn album_get(
 }
 
 #[delete("/album/<id>")]
-async fn album_delete(db: Database, user: DangerousUser, id: String) -> Result<()> {
-    if !user.has_permissions(&[Permission::AlbumDelete]) {
+async fn album_delete(db: Database, user: User, id: String) -> Result<()> {
+    if !user.permissions.contains(&Permission::AlbumDelete) {
         Err(Status::Forbidden)?
     }
 
     db.run(move |conn| -> Result<()> {
+        conn.execute_batch("PRAGMA foreign_keys = ON;")?;
```

Comment:
I don't like specifying this on each endpoint... but `rocket-sync-db-pools` does not allow you to set pool options and there is no easy way around that.

I think I will need to implement my own database system, however this PR was already too big so I will put that in another.